static/panes/cfg-view.ts

```diff
@@ -24,60 +24,47 @@
 
 import * as vis from 'vis-network';
 import _ from 'underscore';
-import {Toggles} from '../widgets/toggles';
 import {ga} from '../analytics';
+import {Toggles} from '../widgets/toggles';
 import TomSelect from 'tom-select';
 import {Container} from 'golden-layout';
-import {CfgState} from './cfg-view.interfaces';
-import {PaneRenaming} from '../widgets/pane-renaming';
+import {CfgOptions, CfgState} from './cfg-view.interfaces';
 import {Hub} from '../hub';
+import {Pane} from './pane';
 
 interface NodeInfo {
     edges: string[];
-    dagEdges: number[];
+    dagEdges: string[];
     index: string;
     id: number;
     level: number;
     state: number;
     inCount: number;
 }
 
-export class Cfg {
-    container: Container;
-    eventHub: any;
-    domRoot: JQuery;
+export class Cfg extends Pane<CfgState> {
     defaultCfgOutput: object;
     llvmCfgPlaceholder: object;
     binaryModeSupport: object;
     savedPos: any;
     savedScale: any;
     needsMove: boolean;
     currentFunc: string;
-    functions: {[key: string]: any};
-    networkOpts: any;
-    cfgVisualiser: any;
-    compilerId: number;
-    _compilerName = '';
-    _editorid?: number;
-    _treeid?: number;
+    functions: Record<string, vis.Data>;
+    networkOpts: vis.Options;
+    cfgVisualiser: vis.Network;
     _binaryFilter: boolean;
     functionPicker: TomSelect;
-    supportsCfg = false;
     toggles: Toggles;
     toggleNavigationButton: JQuery;
     toggleNavigationTitle: string;
     togglePhysicsButton: JQuery;
     togglePhysicsTitle: string;
-    topBar: JQuery;
-    paneName: string;
-    paneRenaming: PaneRenaming;
+    options: Required<CfgOptions>;
```

Comment:
Had to separate `CfgOptions` this to be able to reference it here and have it working.
Ideally, I could have done `{[K in keyof Required<CfgState['options']>]: boolean};` but that returned `{}` instead of the expected `{physics: boolean, navigation: boolean}`